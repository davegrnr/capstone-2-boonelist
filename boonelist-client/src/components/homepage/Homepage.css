span {
	font-size: 18.72px;
}
