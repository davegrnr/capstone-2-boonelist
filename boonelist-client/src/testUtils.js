import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  phone: "111-222-3333",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
