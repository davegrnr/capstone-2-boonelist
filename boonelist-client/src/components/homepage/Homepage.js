import React, { useContext } from "react";
import { Link } from 'react-router-dom'
import UserContext from '../../auth/UserContext'
import './Homepage.css'

/** Site Homepage
 * 
 * Displays welcome message or login/register buttons
 * 
 * Routed at /
 * 
 * Routes -> Homepage
 */

function Homepage(){
    const { currentUser } = useContext(UserContext)

    function loggedOutHomepage(){
        return(
            <div className="Homepage">
                <div className="container">
                    <h1 className="boonelist-heading">Welcome to Boonelist!</h1>
                    <h3>Check out local services offered and items for sale for the Boone, NC and surrounding areas in one convenient place</h3>
                    <button className="btn btn-primary btn-lg mr-3"><Link to="/login" style={{color: "#fff"}}>Login</Link></button>
                    <span> or </span>
                    <button className="btn btn-primary btn-lg ml-3"><Link to="/signup" style={{color: "#fff"}}>Signup</Link></button>
                    <h3 className="mt-2">to get started</h3>
                </div>
            </div>
        )
    }

    function loggedInHomepage(){
        return (
            <div className="Homepage">
                <div className="container">
                    <h1 className="boonelist-heading">Welcome to Boonelist!</h1>
                    <h3>Check out local services offered and items for sale for the Boone, NC and surrounding areas in one convenient place</h3>
                    <h3>OR</h3>
                    <h3>Post your own <Link to="/services/new">service</Link> or <Link to="/sales/new">item</Link></h3>

                </div>

            </div>
        )
    }

    return (
        <div>
            { currentUser ?  loggedInHomepage() : loggedOutHomepage() }
        </div>
    )
}

export default Homepage;