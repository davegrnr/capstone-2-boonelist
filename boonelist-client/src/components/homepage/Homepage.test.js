import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import Homepage from "./Homepage";
import { UserProvider } from "../../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
      <BrowserRouter>
        <UserProvider>
                <Homepage />
        </UserProvider>
      </BrowserRouter>

  );
  expect(asFragment()).toMatchSnapshot();
});
