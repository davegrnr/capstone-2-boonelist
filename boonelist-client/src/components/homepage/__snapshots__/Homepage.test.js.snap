// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Homepage"
    >
      <div
        class="container"
      >
        <h1
          class="boonelist-heading"
        >
          Welcome to Boonelist!
        </h1>
        <h3>
          Check out local services offered and items for sale for the Boone, NC and surrounding areas in one convenient place
        </h3>
        <h3>
          OR
        </h3>
        <h3>
          Post your own 
          <a
            href="/services/new"
          >
            service
          </a>
           or 
          <a
            href="/sales/new"
          >
            item
          </a>
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
